File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/WebPush.java
package work.socialhub.kmastodon.entity;

import com.google.gson.annotations.SerializedName;

/**
 * @author uakihir0
 */
public class WebPush {

    @SerializedName("title")
    private String title;

    @SerializedName("notification_type")
    private String notificationType;
    @SerializedName("notification_id")
    private Long notificationId;

    @SerializedName("preferred_locale")
    private String preferredLocale;
    @SerializedName("access_token")
    private String access_token;

    @SerializedName("icon")
    private String icon;
    @SerializedName("body")
    private String body;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getNotificationType() {
        return notificationType;
    }

    public void setNotificationType(String notificationType) {
        this.notificationType = notificationType;
    }

    public Long getNotificationId() {
        return notificationId;
    }

    public void setNotificationId(Long notificationId) {
        this.notificationId = notificationId;
    }

    public String getPreferredLocale() {
        return preferredLocale;
    }

    public void setPreferredLocale(String preferredLocale) {
        this.preferredLocale = preferredLocale;
    }

    public String getAccess_token() {
        return access_token;
    }

    public void setAccess_token(String access_token) {
        this.access_token = access_token;
    }

    public String getIcon() {
        return icon;
    }

    public void setIcon(String icon) {
        this.icon = icon;
    }

    public String getBody() {
        return body;
    }

    public void setBody(String body) {
        this.body = body;
    }
}

File: core/src/commonMain/kotlin/work/socialhub/kmastodon/entity/WebPush.kt
package work.socialhub.kmastodon.entity

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import kotlin.js.JsExport

@JsExport
@Serializable
class WebPush {

    @SerialName("title")
    var title: String? = null

    @SerialName("notification_type")
    var notificationType: String? = null

    @SerialName("notification_id")
    var notificationId: Int? = null

    @SerialName("preferred_locale")
    var preferredLocale: String? = null

    @SerialName("access_token")
    var access_token: String? = null

    @SerialName("icon")
    var icon: String? = null

    @SerialName("body")
    var body: String? = null
}
